salt/proxy/junos.py

```diff
@@ -118,10 +120,18 @@ def conn():
 
 def alive(opts):
     '''
-    Return the connection status with the remote device.
+    Validate and return the connection status with the remote device.
 
     .. versionadded:: Oxygen
     '''
+    try:
+        thisproxy['conn'].rpc('<file-list><path>/dev/null</path></file-list>')
+
+    except RpcTimeoutError:
+        thisproxy['conn'].connected = False
```

Comment:
I'm not sure you should treat a `RpcTimeoutError` as a failure of `alive()`, but if you choose to do that, you should at least also attempt to close the existing session. This could be done by replacing L131 with:
```
try:
    thisproxy['conn'].close()
except (RpcError, ConnectError):
    # Ignore any PyEZ connection or RPC exceptions when attempting to close.
    pass
```

Note: The call to `close` will set `connected` to `False`, so there's no need to do that on your own.